manifests/install.pp

```diff
@@ -93,7 +93,11 @@
         # any other type of source means we use a package manager (yum) with no 'source' parameter in the
         # package resource below
         $_package_version = $package_version
-        $_provider = 'yum'
```

Comment:
Do we even have to set the provider? Can't we use undef?

Reply:
Leaving it unspecified makes sense in general.. but it's been this way for so long, I didn't want to break anything. This is especially true because when doing an agent upgrade from 6.x to 8.x, we run the 6.x version of the provider to perform the upgrade. If we switch to dnf, then we have to be sure dnf in puppet 6.x works when upgrading to 8.x.